### PR TITLE
Keep groups when replacing nodes

### LINKED ISSUE #7013: Changing type of node removes it from all groups
Tested on Godot 2.1, but pretty positive the issue is there in 2.2+

**Issue description** (what happened, and what was expected):
If a scene was added to a group in the Node viewer (Node is next to Inspector. Group is next to Signals)
and then we try to change the type of the scene (say, static 2d to area2d), the added "Group" is gone! 

Looking for issues in the wrong place, this can lead to headaches if overlooked, barking up the wrong tree is not nice!

**Steps to reproduce:**
Create a scene lets say StaticBody2D
Add it to a group, lets say "Walls"
Change the type of the Scene/Node to Area2D
Grouping is gone.

### PR DESCRIPTION
**Pull request by neikeq**
Fixes #7013

I am not sure if it is fine inside the `keep_data` block.

### CODE REVIEW

File: scene/main/node.cpp [hunk 1]
```diff
@@ -2485,6 +2485,12 @@ void Node::replace_by(Node* p_node,bool p_keep_data) {
 			rd.name=E->get().name;
 			rd.value=get(rd.name);
```
Comment (neikeq) [inline scene/main/node.cpp]: **Not related to this PR**: It looks like this block of code is not finished. Here `rd` should be added to `replace_data`, shouldn't it?

Comment (neikeq) [inline scene/main/node.cpp]: **Not related to this PR**: It looks like this block of code is not finished. Here `rd` should be added to `replace_data`, shouldn't it?

Comment (vnen) [inline scene/main/node.cpp]: Yes, it seems so.

**Comment by neikeq** [comment]
Merging as there does not seem to be any opinion on the `keep_data` doubt.

**Pull request merged**